### PR TITLE
refactor pkg/time

### PR DESCRIPTION
**Pull request by STRRL**
### What problem does this PR solve?

No. It does not relate to any problem. 

### What's changed and how it works?

This PR aims to make it easier to extend TimeChaos Injection.

This PR introducer new interface called `FakeClockInjector` to present a component that could modify the time of the target process. 

```go
type FakeClockInjector interface {
	Inject(pid int) error
	Recover(pid int) error
}
```

Currently, one implementation of `FakeClockInjector` has its corresponding "fake image" as the data which would be poked into the target process. 

For each different `FakeImage`, I suggest building different `FakeClockInjector` based on it. And use the composite pattern if we need to inject more than one fake image into the target process.

We have only one implementation `TimeSkew` for `FakeClockInjector` now.

And it uses full filename(not trim `.o`) as the name of FakeImage, that's a change.

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Chaos Dashboard`
- [x] Need to **cheery-pick to release branches**
	- [x] release-2.1 

### Checklist

Tests

- [x] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### CODE REVIEW

File: pkg/time/asset.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+			rela_section, err := r.Data()
+			if err != nil {
+				log.Error(err, "read rela section")
+				os.Exit(1)
+			}
+			rela_section_reader := bytes.NewReader(rela_section)
+
+			var rela elf.Rela64
+			for rela_section_reader.Len() > 0 {
+				binary.Read(rela_section_reader, elfFile.ByteOrder, &rela)
```
[the remark this answers is not in the document]
Comment (STRRL) [reply pkg/time/asset.go:85]: ignore

**Comment by STRRL** [comment]
PTAL

/cc @YangKeao @Andrewmatilde @ReigenDing

File: pkg/time/asset.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	"embed"
+	"encoding/binary"
+	"os"
+)
+
+//go:embed fakeclock/*.o
+var fakeclock embed.FS
+
+// FakeImage introduce the replacement of VDSO ELF entry and customizable variables.
+// FakeImage could be constructed with
```
Comment (iguoyr) [inline pkg/time/asset.go:30]: This comment seems not completed :D

Comment (STRRL) [reply pkg/time/asset.go]: 🤣

Updated in a1b3a51! PTAL!

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by iguoyr — APPROVED** [review]
LGTM

**Pull request merged**